### PR TITLE
[shopify-buy] Add *vendor* property to Product and Remove duplicated method definition

### PR DESCRIPTION
**Pull request by RosanaRufer**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://help.shopify.com/en/api/custom-storefronts/storefront-api/reference/object/product#vendor
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by Totemika** [comment]
@openminder @straiforos

File: types/shopify-buy/index.d.ts [hunk 1]
```diff
@@ -50,7 +51,6 @@ declare namespace ShopifyBuy {
         fetchWithProducts(id: string): Promise<any[]>; // TODO fix to be a type: Docs: Fetches a single collection by ID on the shop, not including products.
         fetchAll(pageSizeopt?: number): Promise<any[]>; // TODO fix to be a type: Docs: Fetches all collections on the shop, not including products.
         fetchAllWithProducts(): Promise<any[]>; // TODO fix to be a type: DOC: Fetches all collections on the shop, including products.
-        fetchWithProducts(id: string): Promise<any[]>; // TODO fix to be a type: DOC: Fetches all collections on the shop, including products.
```
Comment (Totemika) [inline types/shopify-buy/index.d.ts]: Was duplicated: (lines 50 and line 53)

Comment (straiforos) [reply types/shopify-buy/index.d.ts]: thank you for that

**Review by straiforos — APPROVED** [review]
simple nice code clean up and property addition. Thanks for finding and fixing that.

**Comment by DanielRosenwasser** [comment]
Thanks!

**Comment by Totemika** [comment]
Hi @straiforos how can I run the tests? I could use some help regarding the next steps to raise new pull requests to update these typings with latest js-buy status.

**Comment by straiforos** [comment]
I am currently away from my computer for the next week while on vacation, but from what I recall yarn test should work. If ran in the directory of the shopify types if I recall correctly this is how I was able to do this.

I know the global run for all tests will show the execution of shopify buy. 

I apologize I can not validate that for you. I am fairly confident in that though :)

**Comment by DanielRosenwasser** [comment]
Not on my computer, but I think it's `npm test -- packagename`

**Comment by Totemika** [comment]
Thanks! I run `npm test -- shopify-buy` and it successfully run. 

`yarn test` does also work at the root directory, not in the shopify types, since it looks for a package.json file.

**Pull request merged**